### PR TITLE
Refactor UpdateOAuth2Application

### PR DESCRIPTION
**Pull request by 6543**
as title come up in #11008

### CODE REVIEW

File: models/oauth2_application.go [hunk 1]
```diff
@@ -196,18 +196,34 @@ type UpdateOAuth2ApplicationOptions struct {
...
+	}
+	if app.UID != opts.UserID {
+		return nil, fmt.Errorf("UID missmatch")
 	}
-	if _, err := e.ID(opts.ID).Update(app); err != nil {
+
+	app.Name = opts.Name
+	app.RedirectURIs = opts.RedirectURIs
+
+	if err = updateOAuth2Application(x, app); err != nil {
```
Comment (6543) [inline models/oauth2_application.go]:
```suggestion
	if err = updateOAuth2Application(sess, app); err != nil {
```

**Comment by 6543** [comment]
@guillep2k can you add this to v1.12 pleace?

**Comment by 6543** [comment]
thx @techknowlogick

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
this pull is failing now because of a weard merge behaviour - I'll rebase it

**Review by adelowo — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**